mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java

```diff
@@ -199,6 +209,54 @@ public SendOtpResult sendOtp(String relyingPartyId, String clientId, SendOtpDto
         return new SendOtpResult(sendOtpDto.getTransactionId(), maskedEmailId, maskedMobile);
     }
 
+    private boolean validateKnowledgeBasedAuth(KycAuthRequestDto kycAuthRequestDto,IdentityData identityData){
+        if(CollectionUtils.isEmpty(fieldDetailList)){
+            log.error("KBA field details not configured");
+            throw new MockIdentityException("auth-failed");
+        }
+        String encodedChallenge=kycAuthRequestDto.getKba();
+        try{
+            byte[] decodedBytes = Base64.getUrlDecoder().decode(encodedChallenge);
+            String challenge = new String(decodedBytes, StandardCharsets.UTF_8);
+            Map<String, String> challengeMap = objectMapper.readValue(challenge, Map.class);
+
+            for(Map<String,String> fieldDetail:fieldDetailList){
+                if(challengeMap.containsKey(fieldDetail.get(FIELD_ID_KEY))) {
+                    String challengField = fieldDetail.get(FIELD_ID_KEY);
+                    Object challengeValue = challengeMap.get(challengField);
+
+                    Field field = identityData.getClass().getDeclaredField(challengField);
+                    field.setAccessible(true);
+                    Object fieldValue = field.get(identityData);
+                    if(fieldValue instanceof List){
+                        List<LanguageValue> languageValues = (List<LanguageValue>) fieldValue;
+                        if(!languageValues.get(0).getValue().equals(challengeValue)){
```

Comment:
Let's take the default language from properties for now. Later we should think of taking the input language in the challenge.

Reply:
added language properties